src/lib/common/include/sol-platform.h

```diff
@@ -190,6 +190,53 @@ int sol_platform_get_mount_points(struct sol_ptr_vector *vector);
 int sol_platform_unmount(const char *mpoint, void (*cb)(void *data, const char *mpoint, int error), const void *data);
 
 /**
+ * @brief Gets the hostname.
+ *
+ * @param name The current hostname.
+ *
+ * @return 0 on success, negative errno otherwise.
+ *
+ * @see sol_plataform_set_hostname()
+ */
+int sol_plataform_get_hostname(char **name);
```

Comment:
I'd say we better return the `char *` or hide the memory management complexity and return `const char *`, then internally we can allocate or keep a fixed static buffer, it's easier to use.